src/integrator/cuda/xc_cuda_data.cxx

```diff
@@ -49,12 +51,13 @@ XCCudaData<F>::XCCudaData( size_t _natoms,
   // Allocate static memory with proper alignment
   buffer_adaptor mem( device_ptr, fill_sz );
 
+  LDatoms = util::div_ceil( natoms, cuda::weight_unroll ) * cuda::weight_unroll;
 
   shells_device     = mem.aligned_alloc<Shell<F>>( nshells );
   exc_device        = mem.aligned_alloc<F>( 1 );
   nel_device        = mem.aligned_alloc<F>( 1 );
   acc_scr_device    = mem.aligned_alloc<F>( 1 );
-  rab_device        = mem.aligned_alloc<F>( natoms * natoms );
+  rab_device        = mem.aligned_alloc<F>( LDatoms * natoms, 2 * sizeof(F));
```

Comment:
This aligns the atoms across access boundaries, yes?

Reply:
Yes, this is padding the number of atoms and aligning the memory to 16 bytes, which is required for double2 vector loads.

I do assume that LDatoms is even in order for each row of rab_device to be aligned to 16 bytes. That is part of the reason for the static assert for `weight_unroll`.

Reply:
OK, I think that works in general. I guess that setting it explicitly to align to 16 bytes (or `sizeof(double2)`) might be a bit more clear, just in case we need to look back at this in the future and wonder why a factor of 2 magically appears